### PR TITLE
Re-add vertical input to intent capture

### PR DESCRIPTION
**Pull request by alshakero**
#### Changes proposed in this Pull Request

This PR re-adds the vertical input to the intent capture.
#### Caveat
When the user presses "I don't know" to the vertical, I couldn't figure out how the UI should behave, and it's not spec'd in the design.

#### Testing

Please test as many as possible scenarios of site vertical and site title variations. If you could test on your phone it would be nice too.

### CODE REVIEW

File: client/landing/gutenboarding/onboarding-block/acquire-intent/vertical-select/index.tsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+	return (
+		/* eslint-disable jsx-a11y/click-events-have-key-events */
+		/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+		<form
+			className={ classnames( 'vertical-select', {
+				'vertical-select--with-suggestions': !! suggestions.length && isMobile,
+			} ) }
+		>
+			<label htmlFor="vertical-input">{ __( 'My site is about' ) } </label>
```
Comment (a8ci18n) [inline client/landing/gutenboarding/onboarding-block/acquire-intent/vertical-select/index.tsx:202]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'My site is about <Input />.' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/vertical-input)

**Comment by simison** [comment]
Let's move the underline little lower, so that there's space between the text and underline:

Screenshot 2020-08-07 at 14 14 43


should be more like:

Screenshot 2020-08-07 at 14 15 39

**Comment by simison** [comment]
@olaolusoga hi! could you test this (https://calypso.live/new?branch=add/vertical-input) and give feedback if this looks good?

**Comment by simison** [comment]
On mobile there's a stray arrow:

image

Spacing with underline feels a bit off for mobile, too: 

image

At about 750px width, odd things happen:

image

If user skips the first step, we could just hide the vertical input instead of keeping both visible? Kinda like how it's on mobile.
image

**Comment by olaolusoga** [comment]
Thanks for the ping @simison 

We should match the v2 build for intent capture. Video can be found on the release & result post, P2-pbAok1-16b.

Some notes from comparing the two:
- The input line bottom border is very long, and should be gray when there's no input. (see video)
- When a person types the active text should be underlined with the blue, this controls the length of the bottom border.
- The second field shouldn't have a bottom border, but should keep the placeholder text animation.
- "I don't know" should be in the white bg/gray border button (https://d.pr/i/2gON5M), and when it changes to "Choose a design" (blue) it should be the same width and height (match styling in video)
- The suggestion pop out shouldn't overlay the border line, it should be below it (https://d.pr/i/F6qYpq, see video for reference)

**Comment by razvanpapadopol** [comment]
Few notes after a quick run:

### Accessibility
* Tabbing or pressing _Enter_ doesn't work when both inputs are displayed and you're focused on the first one.
* On mobile there is no way to go back to vertical screen once you advanced to site title screen. I remember there was some arrow.
Screenshot 2020-08-07 at 18 06 47

* Even if the border is full width, the area is not entirely clickable.
Screenshot 2020-08-07 at 18 06 25

* On mobile if you've typed more than 3 characters and Skip button disappears, there is no way to advance to site title screen. There should be a continue button.
Screenshot 2020-08-07 at 18 17 23


### Design
* Skip and continue buttons are not styled consistent with the rest of Gutenboarding. Check `@automattic/onboarding` components.
* Overlapping domain suggestions on mobile
Screenshot 2020-08-07 at 18 03 59

**Comment by alshakero** [comment]
I addressed all feedback, and rebased this and the feature branch. The only discrepancy with the design is the dot after the vertical input. And the "Next" arrow on mobile, I made it show a little lower than it should, it makes the code much much simpler, if this is a deal breaker, please LMK.

**Comment by simison** [comment]
Tiny nitpick, the space between line and the nudge needs to be a bit bigger:

### this PR
Screenshot 2020-08-10 at 18 06 03

### production

Screenshot 2020-08-10 at 18 06 09

**Comment by razvanpapadopol** [comment]
Changing an existing site vertical using the suggestions doesn't work. Tried clicking directly a suggestion and using Enter.
demo: https://cloudup.com/csoMUcDHUmp

**Comment by alshakero** [comment]
Addressed all!

**Comment by simison** [comment]
There's something funky with longer texts when it doesn't fit anymore (note scroller)

Screenshot 2020-08-10 at 19 14 13

I'm not sure what I'd expect in that situation — probably the input get capped in `max-width: 100vw;` or so.

I also observed it produce longer underline somehow in some resolution with longer text:

image

I don't think these are blockers though, and we can follow-up fix tomorrow.

**Comment by simison** [comment]
Note failed `typecheck-strict`

**Comment by simison** [comment]
Note on pad-width the scrollbar and placeholder going over:

Screenshot 2020-08-10 at 19 19 18
 
Scrollbar would be great to get rid of, and in current production version I think it caps at input width so this isn't that biggie.

**Comment by simison** [comment]
Some odd spacing in this resolution:

image

**Comment by razvanpapadopol** [comment]
Something mentioned in https://github.com/Automattic/wp-calypso/pull/44701#issuecomment-670570334 is only partly fixed.
> Tabbing or pressing Enter doesn't work when both inputs are displayed and you're focused on the first one.

Tabbing still doesn't work. 

Also, something that used to work before latest fixes: if you are navigating using arrows through results, you can press _Enter_ to select one of them but you can't proceed anymore with current input by using TAB since this key is now ignored.

**Comment by razvanpapadopol** [comment]
Horizontal scroll issues and zoom on mobile were all fixed with https://github.com/Automattic/wp-calypso/pull/44701/commits/c3e8d3f02c727e7e0349b9a2110b2a633b91df29

One small thing I can see on mobile is the suggestions dropdown overlapping _Next_ arrow.
Screenshot 2020-08-10 at 20 03 07

**Pull request merged**